### PR TITLE
fix: shallow render with fragment at top lvl

### LINKED ISSUE #262: Rendering shallow makes a Fragment into a <p>
Shallow rendering this:
`<><ComponentX /></>`
Creates this:
`<p><ComponentX /></p>`

Is that intended behaviour? Why not just `<ComponentX />`

### PR DESCRIPTION
**Pull request by immitsu**
Fixes #262

### CODE REVIEW

**Comment by remko79** [comment]
Can we get this reviewed and merged please?

File: src/pretty.js [hunk 1]
```diff
@@ -104,10 +104,12 @@ function _renderToStringPretty(
...
 		isComponent = true;
-		if (opts.shallow && (inner || opts.renderRootComponent === false)) {
-			nodeName = getComponentName(nodeName);
-		} else if (nodeName === Fragment) {
-			const children = [];
+
+		let isFragment = nodeName === Fragment,
+			isShallow = opts.shallow && (inner || opts.renderRootComponent === false);
+
+		if (isFragment || (isShallow && isFragment)) {
```
Comment (rschristian) [inline src/pretty.js:111]: This conditional doesn't make much sense

File: src/pretty.js [hunk 2]
```diff
@@ -104,10 +104,12 @@ function _renderToStringPretty(
...
+			let children = [];
```
Comment (rschristian) [inline src/pretty.js:112]: nit: let's keep it as `const`, reduces the diff

```suggestion
			const children = [];
```

File: src/pretty.js [hunk 3]
```diff
@@ -104,10 +104,12 @@ function _renderToStringPretty(
 	// components
 	if (typeof nodeName === 'function') {
 		isComponent = true;
-		if (opts.shallow && (inner || opts.renderRootComponent === false)) {
-			nodeName = getComponentName(nodeName);
-		} else if (nodeName === Fragment) {
-			const children = [];
+
+		let isFragment = nodeName === Fragment,
+			isShallow = opts.shallow && (inner || opts.renderRootComponent === false);
```
Comment (rschristian) [inline src/pretty.js:109]: Can put these right back in the `if()` statements

**Comment by remko79** [comment]
Not sure if I should put this here, but else I can move it to my issue. I think we should only add the check ` && nodeName !== Fragment` on line 107 and omit other changes (except the updated test). The full diff based on current main branch is attached here.
[diff.txt](https://github.com/preactjs/preact-render-to-string/files/15250257/diff.txt)

**Review by rschristian — APPROVED** [review]
Thanks! Sorry for the delay.

**Pull request merged**